fMRIVolume/GenericfMRIVolumeProcessingPipeline.sh

```diff
@@ -342,57 +408,74 @@ fi
+       "$fMRIReferenceImage" \
        "$fMRIFolder"/"$NameOffMRI"_mc \
        "$fMRIFolder"/"$MovementRegressor" \
        "$fMRIFolder"/"$MotionMatrixFolder" \
        "$MotionMatrixPrefix" \
-       "$MotionCorrectionType"
+       "$MotionCorrectionType" \
+       "$fMRIReferenceReg"
 
 # EPI Distortion Correction and EPI to T1w Registration
-log_Msg "EPI Distortion Correction and EPI to T1w Registration"
-
 DCFolderName=DistortionCorrectionAndEPIToT1wReg_FLIRTBBRAndFreeSurferBBRbased
 DCFolder=${fMRIFolder}/${DCFolderName}
 
-if [ -e ${DCFolder} ] ; then
-    ${RUN} rm -r ${DCFolder}
+if [ $fMRIReference = "NONE" ] ; then
+  log_Msg "EPI Distortion Correction and EPI to T1w Registration"
+
+  if [ -e ${DCFolder} ] ; then
+      ${RUN} rm -r ${DCFolder}
+  fi
+  log_Msg "mkdir -p ${DCFolder}"
+  mkdir -p ${DCFolder}
+
+  ${RUN} ${PipelineScripts}/DistortionCorrectionAndEPIToT1wReg_FLIRTBBRAndFreeSurferBBRbased.sh \
+         --workingdir=${DCFolder} \
+         --scoutin=${fMRIFolder}/${ScoutName}_gdc \
+         --t1=${T1wFolder}/${T1wImage} \
+         --t1restore=${T1wFolder}/${T1wRestoreImage} \
+         --t1brain=${T1wFolder}/${T1wRestoreImageBrain} \
+         --fmapmag=${MagnitudeInputName} \
+         --fmapphase=${PhaseInputName} \
+         --fmapgeneralelectric=${GEB0InputName} \
+         --echodiff=${deltaTE} \
+         --SEPhaseNeg=${SpinEchoPhaseEncodeNegative} \
+         --SEPhasePos=${SpinEchoPhaseEncodePositive} \
+         --echospacing=${EchoSpacing} \
+         --unwarpdir=${UnwarpDir} \
+         --owarp=${T1wFolder}/xfms/${fMRI2strOutputTransform} \
+         --biasfield=${T1wFolder}/${BiasField} \
+         --oregim=${fMRIFolder}/${RegOutput} \
+         --freesurferfolder=${T1wFolder} \
+         --freesurfersubjectid=${Subject} \
+         --gdcoeffs=${GradientDistortionCoeffs} \
+         --qaimage=${fMRIFolder}/${QAImage} \
+         --method=${DistortionCorrection} \
+         --topupconfig=${TopupConfig} \
+         --ojacobian=${fMRIFolder}/${JacobianOut} \
+         --dof=${dof} \
+         --fmriname=${NameOffMRI} \
+         --subjectfolder=${SubjectFolder} \
+         --biascorrection=${BiasCorrection} \
+         --usejacobian=${UseJacobian}
+else
+    log_Msg "linking EPI distortion correction and T1 registration from ${fMRIReference}"
+    if [ -e ${DCFolder} ] ; then
+        log_Warn "     ... removing stale link (or preexisiting files)"
+        rm -r ${DCFolder}
+    fi
+    ln -s ${fMRIReference}/${DCFolderName} ${DCFolder}
+ 
+    ${FSLDIR}/bin/imcp ${fMRIReference}/${RegOutput} ${fMRIFolder}/Reference${RegOutput}
+    ${FSLDIR}/bin/imcp ${T1wFolder}/xfms/${fMRIReference}2str ${T1wFolder}/xfms/${fMRI2strOutputTransform}
+    ${FSLDIR}/bin/imcp ${fMRIReferencePath}/${JacobianOut} ${fMRIFolder}/${JacobianOut}
+    ${FSLDIR}/bin/imcp ${fMRIReference}/${QAImage} ${fMRIFolder}/${QAImage}
+    ${FSLDIR}/bin/imcp ${fMRIReference}/${NameOffMRI}_SBRef_nonlin ${fMRIFolder}/${NameOffMRI}_SBRef_nonlin
```

Comment:
This doesn't seem right.  `${NameOffMRI}` for the current BOLD run is not going to be an appropriate name for the `${fMRIReference}`.  The following might work, but I'm not sure what you are trying to accomplish here:
`${FSLDIR}/bin/imcp ${fMRIReferencePath}/${fMRIReference}_SBRef_nonlin ${fMRIFolder}/${fMRIReference}_SBRef_nonlin`
But that's not going to work properly with the code in GenericfMRIVolume

Reply:
That was the intended copy. I corrected it. It will work as in `OneStepResampling` if Reference is specified, a new `ScoutOutput` is not generated, but copied here. There is a note in `OneStepResampling` in that effect (line `256`). But I'll add one here as well.